src/components/PizzaCanvas/PizzaCanvas.js

```diff
@@ -0,0 +1,126 @@
+import Component from '../../Component';
+import {PizzaImagery} from "../../services/PizzaImagery";
+import {CANVAS} from './settings';
+
+/**
+ * Class representing pizza constructed image.
+ * host is a <canvas> element
+ */
+export default class PizzaCanvas extends Component {
+  constructor(props) {
+    super(props);
+    this.ctx = this.host.getContext("2d");
+
+    this.host.width = CANVAS.px.width;
+    this.host.height = CANVAS.px.height;
+  }
+
+  /**
+   * Render view
+   * @returns {HTMLElement|string}
+   */
+  render() {
+    this.ctx.clearRect(0, 0, CANVAS.px.width, CANVAS.px.height);
+    // this.ctx.fillRect(10,10,150,150);
+    console.log(this.name + '.render() state', this.state);
+    this.drawPizzaBase();
+    this.drawIngredients();
+    return '';
+  }
+
+  /**
+   * Draws ingredients
+   */
+  drawIngredients() {
+    const area = CANVAS.areas[this.state.size];
+    console.log(this.name + '.drawIngredients() area', area);
+
+    const ingredientsCount = Object.keys(this.state.ingredients).length;
+
+    const nodes = this._distributeNodesEvenly(area.nodes, ingredientsCount);
+
+    Object.keys(this.state.ingredients).forEach((item, idx) => {
+      PizzaImagery.getImage(item)
+        .then(image => {
+          const coordSet = nodes[idx];
+          for(let i = coordSet.length - 1; i >= 0; i--) {
+            const polar = coordSet[i];
+            this._drawImage(image, polar.r, polar.asimuth, CANVAS.ingredient.width, CANVAS.ingredient.height);
+          }
+        })
+        .catch(rejection => console.log(this.name + '.drawIngredients() REJECTED', rejection));
+    });
+  }
+
+  /**
+   * Distributes array of nodes evenly across count groups
+   * @param {array} nodes [{r,asimuth}...]
+   * @param {number} count
+   * @returns {array} [[{r,asimuth}...]...]
+   * @private
+   */
+  _distributeNodesEvenly(nodes, count) {
```

Comment:
Very cool and elastic distribution function. :+1:  But you should have done pizza ingredients a little bit less spread (less ingredients per layer).